### PR TITLE
Enhance project sturcture

### PR DESCRIPTION
**Pull request by Mmduh-483**

### CODE REVIEW

File: pkg/resources/server.go [hunk 1]
```diff
@@ -315,18 +303,18 @@ func (m *RdmaDevPlugin) cleanup() error {
 }
 
 // GetInfo get info of plugin
-func (m *RdmaDevPlugin) GetInfo(ctx context.Context, rqt *registerapi.InfoRequest) (*registerapi.PluginInfo, error) {
+func (m *resourceServer) GetInfo(ctx context.Context, rqt *registerapi.InfoRequest) (*registerapi.PluginInfo, error) {
 	pluginInfoResponse := &registerapi.PluginInfo{
 		Type:              registerapi.DevicePlugin,
 		Name:              m.resourceName,
-		Endpoint:          filepath.Join(sockDir, m.socketName),
+		Endpoint:          filepath.Join(activeSockDir, m.socketName),
```
Comment (adrianchiris) [inline pkg/resources/server.go]: why was this changed to activeSockDir ?
when creating a resource server i see: sockDir is intialized.

Comment (Mmduh-483) [reply pkg/resources/server.go]: Changed the name to be more clear

Comment (adrianchiris) [reply pkg/resources/server.go]: use m.socketPath
